vortex-array/src/array/implementation.rs

```diff
@@ -243,16 +243,20 @@ impl<A: ArrayImpl + 'static> Array for A {
     fn to_canonical(&self) -> VortexResult<Canonical> {
         let canonical = ArrayCanonicalImpl::_to_canonical(self)?;
         assert_eq!(
-            canonical.as_ref().len(),
             self.len(),
-            "Canonical length mismatch {}",
+            canonical.as_ref().len(),
+            "Canonical length mismatch {}. Expected {} but encoded into {}.",
             self.encoding(),
+            self.len(),
+            canonical.as_ref().len()
         );
```

Comment:
Won't this display 
```
left: <value>
right: <value>
message
```
anyway? We have added these details on couple of places but I'm not certain they actually help.

Reply:
it just doesn't show in the `pytest` output because it seems to only take the first line, but its NBD

Reply:
that's good enough reason to do it imho

Reply:
just to be clear - it still shows in the backtrace, but not in the final test report.

Reply:
Sure, I think this doesn't hurt but I sometimes was confused by double output